horizon/gopal/client.py

```diff
@@ -0,0 +1,55 @@
+import asyncio
+from typing import Optional, Awaitable, Callable
+
+from loguru import logger
+from opal_client import OpalClient
+from opal_client.engine.runner import PolicyEngineRunner
+
+from horizon.config import sidecar_config
+from horizon.gopal.runner import GopalRunner
+
+
+class GOPALClient(OpalClient):
+
+    @staticmethod
+    async def _run_engine_runner(
+        callback: Optional[Callable[[], Awaitable]],
+        engine_runner: PolicyEngineRunner,
+    ):
+        # runs the callback after policy store is up
+        engine_runner.register_process_initial_start_callbacks([callback] if callback else [])
+        async with engine_runner:
+            await engine_runner.wait_until_done()
+
+    async def _run_or_delay_for_engine_runner(
+        self, callback: Callable[[], Awaitable]
+    ):
+        if self.engine_runner:
+            # runs the callback after policy store is up
+            await self._run_engine_runner(callback, self.engine_runner)
```

Comment:
are you still running OPA if GOPAL is enabled?
i thought GOPAL should now manage it.

Reply:
no, the OPAL service manages OPA

Reply:
Didn't want to add all the complexity of managing the OPA service

Reply:
ok for now